odxtools/odxlink.py

```diff
@@ -45,7 +52,7 @@ class OdxLinkId:
 
     #: The name and type of the document fragment to which the
     #: `local_id` is relative to
-    doc_fragments: list[OdxDocFragment]
+    doc_fragments: tuple[OdxDocFragment, ...]
```

Comment:
is there any advantage of changing this to a tuple? I always mentally associate a tuple with a sequence of constant length. While it is true that the document fragments of an individual `OdxLinkId` can be considered constant, between different objects their size can vary...

Reply:
I generally prefer immutable types, if mutability is not necessary. But as i said, that may be wrong 😄

Reply:
I suppose it is a matter of taste: manipulating these objects is kind of a pain IMO (c.f. for example [mksomersaultmodifiedpdx.py](https://github.com/mercedes-benz/odxtools/blob/main/examples/mksomersaultmodifiedpdx.py)), but I see your point from a conceptual perspective. That said, I think this change should factored out into a separate PR.

besides that shouldn't these classes be marked as `frozen=True` if you want to see their objects as immutable?